### PR TITLE
Feat/flat output format

### PR DESCRIPTION
**Pull request by manicprone**
Support for "flat" output format. Includes a new test spec to that focuses on the serializer formats specifically.

Also patched a small issue related to the JSON API Spec (json-api) format where null associations were being added to the results.

### CODE REVIEW

File: test/functional/01_actions/bookshelf/05_serializer_formats.spec.js [hunk 1]
```diff
@@ -0,0 +1,901 @@
...
+      const dataTopLevelKeys = ['attributes', 'id', 'type']
+      expect(Object.keys(payload.data).sort()).toEqual(dataTopLevelKeys)
+
+      // Full "data" response
+      expect(payload).toMatchInlineSnapshot({
+        data: {
+          attributes: {
+            created_at: expect.any(Date),
+            updated_at: expect.any(Date),
+            last_login_at: expect.any(String)
```
Comment (justin-lau) [inline test/functional/01_actions/bookshelf/05_serializer_formats.spec.js:70]: With `vi.setSystemTime()` I think you can safely remove the wildcard dates matcher?

File: src/actions/bookshelf/serializers/flat.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    // Build data package
+    const packageType = (data.attributes) ? 'item' : 'collection'
+    if (packageType === 'item') {
+      json = buildItemPackage(type, data, joint)
+    } else {
+      json = buildCollectionPackage(type, data, joint)
+    }
+  }
+
+  return json
```
Comment (justin-lau) [inline src/actions/bookshelf/serializers/flat.js]: A nitpick suggestion on coding style. It seems cognitively lighter to not have to wrap the code in a 2-layer if-else statement. Same idea would go a long way if applied to the rest of the code to reduce a lot of the indentation.

```suggestion
  return !data
    ? {}
    : packageType === 'item'
    ? buildItemPackage(type, data, joint)
    : buildCollectionPackage(type, data, joint)
```

File: test/functional/01_actions/bookshelf/05_serializer_formats.spec.js (hunk 1 above)
Comment (manicprone) [reply test/functional/01_actions/bookshelf/05_serializer_formats.spec.js:70]: Oh. Right on. I will try that.

File: src/actions/bookshelf/serializers/flat.js (hunk 2 above)
Comment (manicprone) [reply src/actions/bookshelf/serializers/flat.js]: Yep! Perfect. It was copied directly from JSON API serializer, and I just molded it down. You are right.

**Pull request closed without merging**